src/servers/src/tls.rs

```diff
@@ -186,8 +192,8 @@ impl ReloadableTlsServerConfig {
     }
 }
 
-pub fn watch_tls_config(tls_server_config: Arc<ReloadableTlsServerConfig>) -> Result<()> {
-    if tls_server_config.get_tls_option().mode == TlsMode::Disable {
+pub fn check_and_watch_tls_config(tls_server_config: Arc<ReloadableTlsServerConfig>) -> Result<()> {
```

Comment:
```suggestion
pub fn maybe_watch_tls_config(tls_server_config: Arc<ReloadableTlsServerConfig>) -> Result<()> {
```

Naming is hard. But `check_and_watch` infers always watch still.

Reply:
sounds good